packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -302,16 +309,29 @@ export class EngineReportingAgent<TContext = any> {
     if (protobufError) {
       throw new Error(`Error encoding trace: ${protobufError}`);
     }
-    const encodedTrace = Trace.encode(trace).finish();
 
-    const signature = await this.getTraceSignature({
+    const { signature, graphqlValidationFailure } = await this.getTraceSignature({
       queryHash,
       documentAST,
       queryString,
       operationName,
     });
 
-    const statsReportKey = `# ${operationName || '-'}\n${signature}`;
+    const operationNameOrStatic = graphqlValidationFailure ?
+      validationErrorStaticIdentifier : operationName;
+
+    // Optionally store the operation body on validation failure as part of
+    // the trace. This could probably be done within the treebuilder, but
+    // since we only assess the operation signature / document as part of
+    // building up the key to the map, it makes more sense to keep both here
+    // or indicate GraphQL validation failure as an argument to this function.
+    if (graphqlValidationFailure && queryString && this.options.sendOperationDocumentsOnValidationFailure) {
+      trace.operationBodyOnValidationFailure = queryString
```

Comment:
missing semicolon (run prettier or whatever?)

Reply:
No more prettier in this project, but I fixed it manually.